### PR TITLE
Add `storage.truncate()` method

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

**Comment by kasperisager** [comment]
@mafintosh This is ready for review.

File: index.js [hunk 1]
```diff
@@ -76,6 +79,16 @@ module.exports = class RandomAccessStorage extends EventEmitter {
     return NOT_DELETABLE(req)
   }
 
+  truncate (offset, cb) {
+    if (!cb) cb = noop
+    openWritable(this)
+    this.run(new Request(this, TRUNCATE_OP, offset, Infinity, null, false, cb))
```
Comment (mafintosh) [inline index.js]: i think we should set size to `0` here

File: index.js [hunk 2]
```diff
@@ -76,6 +79,16 @@ module.exports = class RandomAccessStorage extends EventEmitter {
...
+  }
+
+  _truncate (req) {
```
Comment (mafintosh) [inline index.js:88]: and then here set size = Infinity for the default impl

**Comment by mafintosh** [comment]
2.2.0

**Pull request merged**